test/python/test_bdd_nexus.py

```diff
@@ -0,0 +1,493 @@
+    with pytest.raises(grpc.RpcError) as error:
+        create_nexus(nexus_uuid, megabytes(128), nexus_children)
+    assert error.value.code() == grpc.StatusCode.INTERNAL
+
+
+@when("destroying the nexus")
+@when("destroying the nexus without unpublishing it")
+def destroying_the_nexus(mayastor_mod, nexus_instance, nexus_uuid, created_nexuses):
+    mayastor_mod[nexus_instance].ms.DestroyNexus(
+        pb.DestroyNexusRequest(uuid=nexus_uuid)
+    )
+    del created_nexuses[nexus_uuid]
+
+
+@when("destroying a nexus that does not exist")
+def destroying_a_nexus_that_does_not_exist(mayastor_mod, nexus_instance):
+    mayastor_mod[nexus_instance].ms.DestroyNexus(
+        pb.DestroyNexusRequest(uuid="e6629036-1376-494d-bbc2-0b6345ab10df")
+    )
+
+
+@when("listing all nexuses", target_fixture="list_nexuses")
+def list_nexuses(mayastor_mod, nexus_instance):
+    return mayastor_mod[nexus_instance].ms.ListNexus(pb.Null()).nexus_list
+
+
+@when(
+    parsers.parse('publishing a nexus via "{protocol}"'), target_fixture="publish_nexus"
+)
+def publish_nexus(mayastor_mod, nexus_instance, nexus_uuid, protocol):
+    mayastor_mod[nexus_instance].ms.PublishNexus(
+        pb.PublishNexusRequest(uuid=nexus_uuid, key="", share=share_type(protocol))
+    )
+
+
+@when("publishing the nexus with the same protocol")
+def publishing_the_nexus_with_the_same_protocol(
+    find_nexus, mayastor_mod, nexus_instance, nexus_uuid
+):
+    nexus = find_nexus(nexus_uuid)
+    mayastor_mod[nexus_instance].ms.PublishNexus(
+        pb.PublishNexusRequest(
+            uuid=nexus_uuid, key="", share=pb.ShareProtocolNexus.NEXUS_NVMF
+        )
+    )
+
+
+@when("attempting to publish the nexus with a different protocol")
+def attempt_to_publish_nexus_with_different_protocol(
+    find_nexus, mayastor_mod, nexus_instance, nexus_uuid
+):
+    nexus = find_nexus(nexus_uuid)
+    with pytest.raises(grpc.RpcError) as error:
+        mayastor_mod[nexus_instance].ms.PublishNexus(
+            pb.PublishNexusRequest(
+                uuid=nexus_uuid, key="", share=pb.ShareProtocolNexus.NEXUS_NBD
+            )
+        )
+    assert error.value.code() == grpc.StatusCode.INVALID_ARGUMENT
+
+
+@when("publishing the nexus using a crypto-key")
+def publish_nexus_with_cryptokey(mayastor_mod, nexus_instance, nexus_uuid):
+    mayastor_mod[nexus_instance].ms.PublishNexus(
+        pb.PublishNexusRequest(
+            uuid=nexus_uuid, key="0123456789123456", share=share_type("nvmf")
+        )
+    )
+
+
+@when("unpublishing the nexus")
+def unpublish_nexus(mayastor_mod, nexus_instance, nexus_uuid):
+    mayastor_mod[nexus_instance].ms.UnpublishNexus(
+        pb.UnpublishNexusRequest(uuid=nexus_uuid)
+    )
+
+
+@when("removing a child")
+def remove_child(mayastor_mod, nexus_instance, nexus_uuid, nexus_children):
+    mayastor_mod[nexus_instance].ms.RemoveChildNexus(
+        pb.RemoveChildNexusRequest(uuid=nexus_uuid, uri=nexus_children[0])
+    )
+
+
+@when("adding the removed child")
```

Comment:
It says adding a removed child but it seems like we are only adding a child.

Reply:
This is a bit confusing when seen out of context. Hopefully it makes sense when you read the entire scenario to which it belongs